### PR TITLE
EVG-19127: Only count failed test names once if match multiple regex filters

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-19127

This fixes a bug where we are counting the same test name more than once if it matches more than one regex filters in the `FindFailedTestResultsSamples` function. This causes duplicate test names and matching test names count greater than the total failed count on the task history UI page. I also removed the consolidation code since that is no longer necessary after https://github.com/evergreen-ci/evergreen/pull/6269 (it is now done client side).

I tested that this works correctly in staging and updated the unit tests.

### CODE REVIEW

**Review by ybrill — COMMENTED** [review]
are we also going to remove the other places where we allow a display task to be specified when fetching test results? (e.g. [this](https://github.com/evergreen-ci/cedar/blob/a0881ac01e4f74545567bc0f7c54d5e5646775c1/model/test_results.go#L716-L718))

**Comment by julianedwards** [comment]
> are we also going to remove the other places where we allow a display task to be specified when fetching test results? (e.g. [this](https://github.com/evergreen-ci/cedar/blob/a0881ac01e4f74545567bc0f7c54d5e5646775c1/model/test_results.go#L716-L718))

Yes! That is coming shortly in a subsequent PR.

File: model/test_results.go [hunk 1]
```diff
@@ -838,21 +778,26 @@ func filterTestSamples(samples []TestResultsSample, regexFilters []string) ([]Te
...
-				if regex.MatchString(name) {
-					filteredNames = append(filteredNames, name)
+	samples := make([]TestResultsSample, len(records))
+	for i, record := range records {
+		samples[i].TaskID = record.Info.TaskID
+		samples[i].Execution = record.Info.Execution
+
+		samples[i].TotalFailedTestNames = len(record.FailedTestsSample)
+		for _, testName := range record.FailedTestsSample {
+			match := true
```
[the remark this answers is not in the document]
Comment (julianedwards) [reply model/test_results.go:788]: If no regexes are provided we should match everything. This was always the behavior (see L773 of the original code).

[the remark this answers is not in the document]
Comment (ybrill) [reply model/test_results.go:788]: oh! I see why I missed that.

**Review by ybrill — APPROVED** [review]

File: model/test_results.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianedwards) [reply model/test_results.go:788]: Yeah this code is a bit complex! haha

**Pull request merged**